# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webidl-api/ExtensionEventListener.cpp
# Commit: 7802e5f1bd13
# Full Hash: 7802e5f1bd1395f19f2d78636787c41f64192bd7
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-04-19 15:22:44
# Regressor Bug: 1769913
# File Overlap Count: 3
# Description:
#   Bug 1769913 - P3 Remove WorkerRunnable::mWorkerPrivate. r=dom-worker-reviewers,asuth
#   
#   WorkerRunnable no longer keeps a raw pointer(mWorkerPrivate) for the associated WorkerPrivate in this patch.
#   Removing the WorkerRunnable::mWorkerPrivate needs to fix the following problems.
#   
# ==============================================================================

diff -r ffc69aa8f2c3 -r 7802e5f1bd13 toolkit/components/extensions/webidl-api/ExtensionEventListener.cpp
--- a/toolkit/components/extensions/webidl-api/ExtensionEventListener.cpp	Fri Apr 19 09:41:57 2024 +0000
+++ b/toolkit/components/extensions/webidl-api/ExtensionEventListener.cpp	Fri Apr 19 09:41:58 2024 +0000
@@ -299,7 +299,7 @@
   RefPtr<ExtensionListenerCallWorkerRunnable> runnable =
       new ExtensionListenerCallWorkerRunnable(this, std::move(argsHolder),
                                               aCallOptions, retPromise);
-  runnable->Dispatch();
+  runnable->Dispatch(GetWorkerPrivate());
   retPromise.forget(aPromiseResult);
 
   return NS_OK;
@@ -332,7 +332,6 @@
     JSContext* aCx, dom::WorkerPrivate* aWorkerPrivate) {
   MOZ_ASSERT(aWorkerPrivate);
   aWorkerPrivate->AssertIsOnWorkerThread();
-  MOZ_ASSERT(aWorkerPrivate == mWorkerPrivate);
   auto global = mListener->GetGlobalObject();
   if (NS_WARN_IF(!global)) {
     return true;